src/main/scala/com/fulcrumgenomics/phasing/HapCutToVcf.scala

```diff
@@ -0,0 +1,413 @@
+      offset       = offset,
+      haplotype1   = tokens(1).toInt,
+      haplotype2   = tokens(2).toInt,
+      contig       = tokens(3),
+      pos          = tokens(4).toInt,
+      ref          = tokens(5),
+      alts         = tokens(6),
+      genotype     = tokens(7),
+      info         = HapCutCallGenotypeInfo(tokens(8)),
+      firstInBlock = blockOffset == offset
+    )
+  }
+
+  /** Converts the HapCut variant representation to a VariantContext. */
+  def toVariantContext(call: HapCutCall, sampleName: String): VariantContext = {
+    // Parse the alleles
+    val refAllele = Allele.create(call.ref, true)
+    val altAlleles = call.alts.split(",").map { alt =>
+      Allele.create(alt, false)
+    }.toSeq
+    val alleles = refAllele +: altAlleles
+    val allelesCollection = alleles.asJavaCollection
+
+    // Assumes the genotype is before the first ":" token
+    val genotypeAlleles = call.genotype.split(":").head.split("/").map(_.toInt).map(i => alleles(i))
+    val genotype = HapCutCallGenotypeInfo.addTo(new GenotypeBuilder(sampleName, genotypeAlleles.toList), call.info).make()
+
+    // build the context, and make sure to recompute the end
+    new VariantContextBuilder(
+      s"${call.offset}",
+      call.contig,
+      call.pos.toLong,
+      call.pos.toLong,
+      allelesCollection
+    )
+      .computeEndFromAlleles(seqAsJavaList(alleles.toSeq), call.pos)
+      .genotypes(genotype).make()
+  }
+}
+
+/** Reads phased blocks and variants from a HapCut output file, returning a HapCut call for each input variant call to
+  * HapCut.  If the variant was not phased by HapCut, None is returned for that variant (typically it is not present in
+  * the HapCut output).  Variants that were not phased after the last HapCut phased variant will not be returned (hasNext
+  * will be false), so must be treated specially for now.
+  */
+private[phasing] class HapCutReader(input: FilePath) extends Iterator[Option[HapCutCall]] {
+  private val reader = Io.toReader(input)
+
+  private var block: Option[HapCutBlockInfo] = None
+  private var calls: Iterator[Option[HapCutCall]] = Iterator.empty
+
+  // initialize
+  this.advance()
+
+  def hasNext(): Boolean = {
+    this.block match {
+      case None => this.advance()
+      case Some(b) if calls.hasNext => true
+      case Some(b) => this.advance()
+    }
+  }
+
+  def next(): Option[HapCutCall] = {
+    if (!hasNext()) throw new NoSuchElementException("Calling next() when hasNext() is false")
+    calls.next()
+  }
+
+  def close(): Unit = {
+    this.reader.close()
+  }
+
+  /** Reads in the next block of data */
+  private def advance(): Boolean = {
+    val blockLine = reader.readLine() match {
+      case null => return false
+      case line => line.trim
+    }
+    if (blockLine == "********") return this.advance()
+    val lastOffset = this.block match {
+      case None => 0
+      case Some(b) => b.offset + b.len - 1
+    }
+    val currentBlock = HapCutBlockInfo(blockLine, -1)
```

Comment:
Huh, I don't understand why it is harder to follow, since having it in the case class allows me not to be distracted by all the parsing code in this simple method.  What in the case class would you have liked seen here in regards to the parsing code?